cpp/src/bitmask/null_mask.cu

```diff
@@ -95,19 +95,52 @@ rmm::device_buffer create_null_mask(size_type size, mask_state state,
   return mask;
 }
 
-//Set pre-allocated null mask to:
-//all entries to valid, if valid_flag==true,
-//or null, otherwise;
-void set_null_mask(bitmask_type* bitmask,
-                   size_type size, bool valid,
-                   cudaStream_t stream)
-{
-  if (bitmask != nullptr) {
-    size_type mask_size = bitmask_allocation_size_bytes(size);
+__global__ void set_null_mask_kernel(bitmask_type *__restrict__ destination,
+                                     size_type begin_bit, size_type end_bit,
+                                     bool valid,
+                                     size_type number_of_mask_words) {
+  auto x = destination + word_index(begin_bit);
+  const auto last_word = word_index(end_bit) - word_index(begin_bit);
+  bitmask_type fill_value = (valid == true) ? 0xffffffff : 0x00;
 
-    uint8_t fill_value = (valid == true) ? 0xff : 0x00;
-    CUDA_TRY(cudaMemsetAsync(bitmask,
-                             fill_value, mask_size, stream));
+  for (size_type destination_word_index = threadIdx.x + blockIdx.x * blockDim.x;
+       destination_word_index < number_of_mask_words;
+       destination_word_index += blockDim.x * gridDim.x) {
+    if (destination_word_index == 0 || destination_word_index == last_word) {
+      bitmask_type mask = 0xffffffff;
+      if (destination_word_index == 0) {
+        mask = (0xffffffff << intra_word_index(begin_bit));
```

Comment:
I think the functions `set_least_significant_bits` and `set_most_significant_bits` could be used here to make the logic a little easier to understand: https://github.com/rapidsai/cudf/blob/branch-0.13/cpp/include/cudf/utilities/bit.hpp#L116

Reply:
I implemented with similar logic (not using these functions) but then I realized it involves few extra operations (few subtractions). So, I went with fully logical operations. 
I replaced with these functions for easier understanding.